python/ray/serialization.py

```diff
@@ -242,8 +237,14 @@ def deserialize_objects(self, data_metadata_pairs, object_refs):
                                                 data_metadata_pairs):
             assert self.get_outer_object_ref() is None
             self.set_outer_object_ref(object_ref)
-            results.append(
-                self._deserialize_object(data, metadata, object_ref))
+            obj = None
+            try:
+                obj = self._deserialize_object(data, metadata, object_ref)
+            except Exception as e:
+                logger.exception(e)
+                obj = RaySystemError(e, traceback.format_exc())
+            assert obj is not None
```

Comment:
Will this fail if the object's value is actually None?

Reply:
Ah, good point. It will. Let me remove this line.